guides/puppet-web-apps/html-views/README.md

```diff
@@ -0,0 +1,98 @@
+2. Through an **insertion point**. An insertion point is a combination of a `Self.GET` call, which acts as the partial view-model insertion point, and a [`<starcounter-include>`](https://github.com/Polyjuice/starcounter-include) tag, which acts as a partial HTML view insertion point. The `<starcounter-include>` tag determines where in the DOM tree the partial HTML view should be rendered.
+
+#### Examples
+
+The examples use [Web Component](/guides/puppet-web-apps/introduction-to-web-components) features and work best in Google Chrome. Other browsers are supported using [polyfills](http://webcomponents.org/polyfills/).
+
+##### Creating the Handler
+
+No matter if the partial is accessed through an HTTP request from the browser or a `Self.GET` call, there always needs to be a handler that deals with the request. This handler simply returns a partial containing the JSON and HTML. The handler, in its simplest form, looks something like this:
+
+```cs
+Handle.GET("/your/partial/url", () => 
+{
+    return new YourPartialPage();
+}
+```
+
+Keep in mind that this only returns the JSON and HTML if the app is using the `HtmlFromJsonProvider` and `PartialToStandaloneHtmlProvider` [middlerware](/guides/network/middleware/).
+
+##### Adding the Insertion Point
+
+It is now possible to create an insertion point for this partial. The way to do this is by attaching the partial to a parent partial using `Self.GET`. For example:
+
+```cs
+mainPage.SubPage = Self.GET("/your/partial/url");
+```
+
+In the partial HTML view for the `mainPage` above, the HTML from the `SubPage` partial can be stamped in like so:
+
+```html
+<starcounter-include partial="{{model.SubPage}}"></starcounter-include>
+```
+
+##### Partial HTML View Example
+
+A partial HTML view may look something like this:
+
+{% raw %}
+```html
+<!--
+    Load you dependencies: <script>s, HTML Imports, CSS stylesheets etc.
+    Those dependencies will only be executed once when the partial is imported
+-->
+<!-- For example, to use Polymer's dom-bind custom element: -->
+<link rel="import" href="/sys/polymer/polymer.html" />
+<style>
+.myapp-address-entry__name {
+    font-weight: bold;
+}
+</style>
+<!--
+    Everything from this template will be stamped into the parent page's DOM,
+    and <script>s will be executed for every instance of this partial.
+    All child nodes will receive a `.model` property with JSON view model,
+    which is automatically bound to the server-side.
+-->
+<template>
+    <template is="dom-bind">
+        <!--
+            Now, the double curly brace syntax "{{}}" can be used for two-way data bindings
+            from the HTML view to serverside. For example:
+         -->
+        <h2 class="myapp-address-entry__name">{{model.FullName}}</h2>
+        <h4>Address</h4>
+        <starcounter-include partial="{{model.Address}}"></starcounter-include>
+    </template>
+</template>
```

Comment:
In the [HTML View Guidelines](https://docs.starcounter.io/guides/puppet-web-apps/html-view-guidelines/) we say that all the content goes into the `dom-bind`. From what I understood that is the preferred approach since that allows the developer to place the elements in the order they appear in the view. Otherwise the elements using bindings and those that don't will be split.